File: Easy/nice_angles.py
data = ["330.39991833", "0.001", "14.64530319", "0.25", "254.16991217"]

#import sys
#test_cases = open(sys.argv[1], 'r')
#data = test_cases.readlines()
for test in data:
	test = test.rstrip()
	angle = float(test)
	whole = int(angle)
	frac1 = int((angle % 1) * 60)
	frac2 = int((((angle % 1) * 60) - frac1) * 60)
	A = str(whole) + "."
	f1 = str(frac1)
	if len(f1) == 1:
		A = A + "0" + f1 + "'"
	else:
		A = A + f1 + "'"
	
	f2 = str(frac2)
	if len(f2) == 1:
		A = A + "0" + f2 + '"'
	else:
		A = A + f2 + '"'
	
	print A
	
#test_cases.close()

File: Easy/nice_angles.r
test.cases <- c("330.39991833", "0.001",
				"14.64530319", "0.25",
				"254.16991217")

#args <- commandArgs(trailingOnly=TRUE)
#test.cases <- strsplit(readLines(args[[1]], warn=FALSE), '\n')
cat(sapply(test.cases, function(test) {
	if (length(test) > 0) {
		angle <- as.numeric(test)
		whole <- floor(angle)
		frac_1 <- floor((angle - whole)*60)
		frac_2 <- floor((((angle - whole) * 60) - frac_1) * 60)
		D <- as.character(whole)
		M <- as.character(frac_1)
		if (nchar(M) == 1) {
			M <- paste(c("0", M), sep = "", collapse = "")
		}
		S <- as.character(frac_2)
		if (nchar(S) == 1) {
			S <- paste(c("0", S), sep = "", collapse = "")
		}
		paste(c(D, ".", M, "'", S, '"'), sep = "", collapse = "")
	}
}), sep = "\n")